source/tests/SGM/SGMEnergyPositionTest.cpp

```diff
@@ -0,0 +1,168 @@
+
+    // Iron L Edge:
+	testIndividualRequestEnergy("Iron L Edge Start", 690.0, SGMGratingSupport::MediumGrating, -295012.44, 16.28355603, 22.1484378, -42938.76);
+	testIndividualRequestEnergy("Iron L Edge Middle", 720.0, SGMGratingSupport::MediumGrating, -282720.26, 16.35620468, 22.64942635, -27022.08);
+	testIndividualRequestEnergy("Iron L Edge End", 750.0, SGMGratingSupport::MediumGrating, -271411.45, 23.59061352, 23.14434469, -31177.09);
+
+    // Cobalt L Edge:
+    // NOTE : Cobalt L Edge Start is the same as Iron L Edge End, so I'm omitting it
+	testIndividualRequestEnergy("Cobalt L Edge Middle", 780.0, SGMGratingSupport::MediumGrating, -260972.54, 36.92797993, 23.63452204, -25388.09);
+	testIndividualRequestEnergy("Cobalt L Edge End", 820.0, SGMGratingSupport::MediumGrating, -248242.18, 62.74685386, 24.28288187, -17730.95);
+
+    //Nickel L Edge:
+	testIndividualRequestEnergy("Nickel L Edge Start", 840.0, SGMGratingSupport::MediumGrating,  -242331.65, 78.64026908, 24.60563035, -13919.29);
+	testIndividualRequestEnergy("Nickel L Edge Middle", 865.0, SGMGratingSupport::MediumGrating,  -235327.84, 100.965777, 25.00841512, -9162.40);
+	testIndividualRequestEnergy("Nickel L Edge End", 890.0, SGMGratingSupport::MediumGrating,  -228717.51, 125.7613523, 25.41110095, -4406.68);
+
+    // Magnesium K Edge:
+	testIndividualRequestEnergy("Magnesium K Edge Start", 1290.0, SGMGratingSupport::HighGrating, -243735.55, 91.20661474, 32.4753518, 79022.19);
+	testIndividualRequestEnergy("Magnesium K Edge Middle", 1315.0, SGMGratingSupport::HighGrating, -239101.80, 105.3760676, 33.0118301, 85358.00);
+	testIndividualRequestEnergy("Magnesium K Edge End", 1340.0, SGMGratingSupport::HighGrating, -234640.94, 120.6471271, 33.57032489, 91953.83);
+
+    // Aluminium K Edge
+	testIndividualRequestEnergy("Aluminium K Edge Start", 1550.0, SGMGratingSupport::HighGrating, -202850.88, 283.8214953, 19.05357132, -79489.15);
+	testIndividualRequestEnergy("Aluminium K Edge Middle", 1580.0, SGMGratingSupport::HighGrating,  -198999.28, 311.2307212, 19.24520489, -77225.96);
+	testIndividualRequestEnergy("Aluminium K Edge End", 1610.0, SGMGratingSupport::HighGrating, -195291.22, 339.467824, 19.43479335, -74986.92);
+
+    // Silicon K Edge
+	testIndividualRequestEnergy("Silicon K Edge Start", 1830.0, SGMGratingSupport::HighGrating, -171813.59, 567.5737302, 20.77155529, -59199.75);
+	testIndividualRequestEnergy("Silicon K Edge Middle", 1860.0, SGMGratingSupport::HighGrating, -169042.40, 601.0803091, 20.94762194, -57120.40);
+	testIndividualRequestEnergy("Silicon K Edge End", 1890.0, SGMGratingSupport::HighGrating, -166359.19, 635.0727254, 21.12242885, -55055.93);
+}
+
+void SGMEnergyPositionTest::testResultantEnergy()
+{
+    SGMEnergyPosition testEnergyPosition(240, SGMGratingSupport::LowGrating);
+
+    if(!qFuzzyCompare(testEnergyPosition.resultantEnergy(), 240)) {
```

Comment:
It does a 'close enough' comparison between two floating point values. I only use it here because its a test class (it has some issues if either value can possibly be zero, which makes it pretty useless in other situations).

this call is essentially equivalent to 

qAbs(testEnergyPosition.resultantEnergy() - 240) < someToleranceValue

(I'm not sure what someToleranceValue qt uses)